wren/compute/materialization_manager.go

```diff
@@ -110,6 +119,77 @@ func (m *materializationManager) GetMaterializationStatus(ctx context.Context, m
 	return statusResponse.Progress, nil
 }
 
+// ReconcileMaterializations reconciles the materializations in the database with the materializations on the compute backend
+// After running this function, all materializations in the database will be running on the compute backend
+// and all deleted materializations will be stopped
+func (m *materializationManager) ReconcileMaterializations(ctx context.Context) error {
+	subLogger := log.Ctx(ctx).With().Str("method", "manager.ReconcileMaterializations").Logger()
+
+	allStreamMaterializations, err := m.materializationClient.GetAllMaterializationsBySourceType(ctx, materialization.SourceTypeStreams)
+	if err != nil {
+		subLogger.Error().Err(err).Msg("failed to get all stream materializations")
+		return err
+	}
+
+	// find all materializations in the database and start any that are not running
+	// we keep a map of materialization_name=>nil to keep track of which materializations are running
+	newRunningMaterializations := make(map[string]interface{})
+	for _, streamMaterialization := range allStreamMaterializations {
+		materializationID := streamMaterialization.ID.String()
+		owner := streamMaterialization.Edges.Owner
+
+		isRunning := false
+		// check to see if the materialization was running in the previous iteration
+		if _, found := m.runningMaterializations[materializationID]; found {
+			//verify that the materialization is still running
+			progressInfo, err := m.GetMaterializationStatus(ctx, materializationID)
+			if err != nil {
+				log.Error().Err(err).Str("id", materializationID).Msg("failed to get materialization status")
+			}
+			isRunning = progressInfo != nil
```

Comment:
Let's make a note to look back at this logic later once Sparrow is done. We may want the ability to `pause` a materialization, or it may have `failed` and we want to report the failure reason (but we'll still want the progress to see what's been produced up til failure). 

This probably leads me to think we'll have a `MaterializationStatusResponse` that wraps the `ProgressInfo` and includes some materialization-specific state as well.